src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -596,6 +600,9 @@ public static void FoundCertWithPrivateKey(this ILogger<HttpsConnectionMiddlewar
     [LoggerMessage(8, LogLevel.Debug, "Failed to open certificate store {StoreName}.", EventName = "FailToOpenStore")]
     public static partial void FailedToOpenStore(this ILogger<HttpsConnectionMiddleware> logger, string? storeName, Exception exception);
 
+    [LoggerMessage(9, LogLevel.Information, "Certificate with thumbprint {Thumbprint} lacks the subjectAlternativeName (SAN) extension and may not be accepted by browsers.", EventName = "NoSubjectAlternativeName")]
```

Comment:
How unique do IDs have to be?  I would have guessed per-logger-type, but the non-generic `ILogger` extension methods seem to reuse them freely.

Reply:
Uniqueness is the `ILogger` name +  logger message ID (which is ID and string)

Usually, the logger used by a type has the same name as the type, e.g. `Microsoft.AspNetCore.Kestrel.HttpConnectionMiddleware`. Then all the logs defined here can have their own IDs that start from 1.

tldr as long as they are unique in this file you should be fine.

Reply:
But in this case since you're passing in KestrelTrace you're breaking that assumption. Should probably try to get an `ILogger<HttpsConnectionMiddleware>` in `BindAsync` and pass that in.

Reply:
Yes. The KestrelTrace logger won't have the name `Microsoft.AspNetCore.Kestrel.HttpConnectionMiddleware`. It has its own `ILogger` name and will already have a log message with ID=9. So that won't be unique.

Reply:
> What would be the consequences of sharing a logger across endpoints?

Shouldn't be any, they are already effectively Singletons in DI.

I vote for whatever option gives us an `ILogger<HttpsConnectionMiddleware>`.

Reply:
@BrennanConroy Tell me more about the expectation that they'll be singletons?  That does seem to be the case for the [default impl](https://source.dot.net/#Microsoft.Extensions.Logging/LoggerFactory.cs,143), but it's not clear from the name ("factory", rather than "provider") or the [doc comment](https://source.dot.net/#Microsoft.Extensions.Logging.Abstractions/ILoggerFactory.cs,15)
```csharp
        /// <summary>
        /// Creates a new <see cref="ILogger"/> instance.
        /// </summary>
        /// <param name="categoryName">The category name for messages produced by the logger.</param>
        /// <returns>The <see cref="ILogger"/>.</returns>
        ILogger CreateLogger(string categoryName);
```

Reply:
https://source.dot.net/#Microsoft.Extensions.Logging/LoggingServiceCollectionExtensions.cs,38
and
https://source.dot.net/#Microsoft.Extensions.Logging/LoggingServiceCollectionExtensions.cs,39
is where logging gets added for use in ASP.NET Core apps. And AddLogging is called from the host builders.

If you use `ILoggerFactory` it has a cache of loggers that it creates/gets from:
https://source.dot.net/#Microsoft.Extensions.Logging/LoggerFactory.cs,143 and the `ILogger` impl we use, actually just calls the `ILoggerFactory` impl: https://source.dot.net/#Microsoft.Extensions.Logging.Abstractions/LoggerT.cs,26

So at the end of the day it's all singleton loggers.